src/main/java/nus/climods/logic/commands/AddCommand.java

```diff
@@ -38,19 +42,35 @@ public AddCommand(String moduleCode) {
         toAdd = moduleCode;
     }
 
+    private Module getModule(Model model, String moduleCode) throws CommandException {
+        Optional<Module> optionalModule = model.getModuleList().getListModule(moduleCode);
```

Comment:
oh the implementation for UserModule has been changed. It no longer take in a model because Model needs the UserModuleList to initiate and having UserModule to take in Model creates a cyclical dependency . So the check will not be in UserModule